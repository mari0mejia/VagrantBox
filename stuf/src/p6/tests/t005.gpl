int theend=2;

termination {
  print("First termination block. Confirming is called after an GPL exit() call");
}




termination {
  print("  This is one termination block. theend=" + theend);
  theend=12;
  print("  Now theend=" + theend);
}

termination {
  print("  There can be more than one termination block. theend=" + theend);
  theend=22;
  print("  And now theend=" + theend);
}

on akey {
  print("Inside event block for akey. About to call exit()");
  exit(1);
}

on akey {
  print("Should never reach this");
}
