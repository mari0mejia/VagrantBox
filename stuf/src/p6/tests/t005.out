gpl.cpp::main()
  input file(tests/t005.gpl)
  random seed(42)
  read_keypresses_from_standard_input(true)
  dump_pixels(false)
  symbol_table(true)
  graphics(true)

gpl.cpp::main() Calling yyparse()


gpl.cpp::main() after call to yyparse().


Printing the symbol table from main()
int theend = 2
gpl.cpp::main() Calling window->initialize().


gpl.cpp::main() Passing control to window->main_loop().
print[23]: Inside event block for akey. About to call exit()
print[4]: First termination block. Confirming is called after an GPL exit() call
print[11]:   This is one termination block. theend=2
print[13]:   Now theend=12
print[17]:   There can be more than one termination block. theend=12
print[19]:   And now theend=22
gpl[24]: exit(1)
